runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/util/BoundedQueueExecutor.java

```diff
@@ -48,6 +49,7 @@ public BoundedQueueExecutor(
       int maximumElementsOutstanding,
       long maximumBytesOutstanding,
       ThreadFactory threadFactory) {
+    this.maximumThreadCount.set(maximumPoolSize);
```

Comment:
how about 
this.maximumThreadCount = new AtomicInteger(maximumPoolSize);
instead of assignment above

Reply:
Changed to int and use synchronized instead.